src/writers/dir_builder/mod.rs

```diff
@@ -139,6 +139,16 @@ impl DirectoryBuilder {
 
     /// Runs `cargo new` for the path to be generated
     fn run_cargo_new(project_name: &String) {
-        Command::new("cargo new").args(&[project_name]);
+        // check if the dir exists, if so, just return early
+        if Path::new(project_name).exists() {
+            return;
+        }
+        // make the directory first
+        DirBuilder::new().create(project_name).unwrap();
+
+        Command::new("cargo")
+            .args(&["init", project_name])
+            .output()
+            .unwrap();
```

Comment:
running `cargo new` introduces a really strange race condition where the `Cargo.toml` file is generated _sometimes_ instead of always.

Making the dir and then just running `cargo init <dir>` is 100% consistent (from my testing) although I don't know why the other method isn't.

Reply:
Maybe its asynchronous and we need some sort of await statement?

Reply:
Nvm, cargo init is the right thing to do regardless